### PR TITLE
Fix rules pattern and samples

### PR DESCRIPTION
**Pull request by babenek**
## Description

- Update rigth bariers for rules
- Fix length of sample for "Square OAuth Secret"
- 4 Google Multi credentials in benchmark ( "AIza...") were removed with fixed rule

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
 SAMPLES_FILES_COUNT: int = 101
 
 # credentials count after scan
-SAMPLES_CRED_COUNT: int = 97
-SAMPLES_CRED_LINE_COUNT: int = 101
+SAMPLES_CRED_COUNT: int = 96
+SAMPLES_CRED_LINE_COUNT: int = 100
```
Comment (csh519) [inline tests/__init__.py:8]: It seems there is affected credential which not detected with this changes.
Could you please check this?

File: credsweeper/rules/config.yaml [hunk 2]
```diff
@@ -97,7 +97,7 @@
   severity: high
   type: pattern
   values:
-    - (?P<value>EAACEdEose0cBA[0-9A-Za-z]+)
+    - (^|[^0-9A-Za-z])(?P<value>EAACEdEose0cBA[0-9A-Za-z]+)
```
Comment (csh519) [inline credsweeper/rules/config.yaml:100]: Isn't it need to be changed the right value?
I just wonder how the right value is added or not.

File: tests/__init__.py (hunk 1 above)
Comment (babenek) [reply tests/__init__.py]: That it was fixed. Picatic test sample matched stripe apki key without right barier.
The values are for scan without ml, so i created "GoldenTest"
https://github.com/Samsung/CredSweeper/pull/360

File: credsweeper/rules/config.yaml (hunk 2 above)
Comment (babenek) [reply credsweeper/rules/config.yaml:100]: The key pattern has no concrete size - rigth barier is extra.

File: credsweeper/app.py [hunk 3]
```diff
@@ -150,7 +150,7 @@ def _get_config_dict(
     # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
     def _use_ml_validation(self) -> bool:
-        if isinstance(self.ml_threshold, float) and self.ml_threshold <= 0:
+        if (isinstance(self.ml_threshold, float) or isinstance(self.ml_threshold, int)) and 0 >= self.ml_threshold:
```
Comment (xDizzix) [inline credsweeper/app.py:153]:
```suggestion
        if isinstance(self.ml_threshold, (float, int)) and 0 >= self.ml_threshold:
```

**Review by xDizzix — APPROVED** [review]
Looks good.
Hope You didn't use real credentials in the test samples.

File: tests/test_main.py [hunk 4]
```diff
@@ -752,13 +750,15 @@ def prepare(report: List[Dict[str, Any]]):
         # do not use parametrised tests with unittests
         self.maxDiff = 65536
         # instead the config file is used
-        with open(TESTS_PATH / "data" / ".cfg.json", "r") as f:
-            config = json.load(f)
         with tempfile.TemporaryDirectory() as tmp_dir:
-            for cfg in config:
+            for cfg in DATA_TEST_CFG:
                 # important key in .cfg.json is "json_filename"
```
Comment (csh519) [inline tests/test_main.py:755]: It seems it can be removed :)

```suggestion
```

File: tests/__init__.py (hunk 1 above)
Comment (csh519) [reply tests/__init__.py]: I understood. Thank you for explanation.

File: credsweeper/rules/config.yaml (hunk 2 above)
Comment (csh519) [reply credsweeper/rules/config.yaml:100]: Ah, I see :)

File: credsweeper/rules/config.yaml [hunk 5]
```diff
@@ -174,7 +174,7 @@
   severity: high
   type: pattern
   values:
-    - (?P<value>(?i)heroku(.{0,20})?[0-9a-f]{8}(-[0-9a-f]{4})+-[0-9a-f]{12})
+    - (?P<value>(?i)heroku(.{0,20})?[0-9a-f]{8}(-[0-9a-f]{4})+-[0-9a-f]{12})([^0-9a-z]|$)
   filter_type: GeneralPattern
   use_ml: true
```
Comment (csh519) [inline credsweeper/rules/config.yaml:179]: Could you explain me why this right value is different with others?

- This one : `([^0-9a-z]|$)`
- Others : `([^=0-9A-Za-z_/+-]|$)`

Comment (babenek) [reply credsweeper/rules/config.yaml:179]: Something were missed. I'll check

File: tests/test_main.py (hunk 4 above)
Comment (babenek) [reply tests/test_main.py:755]: applied in https://github.com/Samsung/CredSweeper/pull/360 - the parent of this PR

Comment (csh519) [reply tests/test_main.py]: Thank you for a quick fix. I checked.

**Review by csh519 — APPROVED** [review]
Thank you for fix patterns of rules and samples.

LGTM 👍

**Review by xDizzix — APPROVED** [review]
+

**Pull request merged**